openquake/calculators/postproc/compute_rtgm.py

```diff
@@ -326,6 +340,284 @@ def get_asce41(dstore, mce, facts):
+            total_poe.append(wp)
+        else:
+            for wp_i in wp:
+                wsp.append(wp_i)
+            total_poe.append([sum(t) for t in np.array(wsp).T])
+    grouped_2['poes'] = total_poe
+    return grouped_2, imtls_dict
+
+
+def _find_sources(df, imtls_dict, imt_list, rtgm_probmce, mean_hcurve, dstore):
+
+    fig, ax = plt.subplots(3, figsize=(8, 15))
+
+    # identify the sources that have a contribution > than fact (here 10%) of
+    # the largest contributor;
+    fact = 0.1
+
+    for m, imt in enumerate(imt_list):
+        out_contr_all = []
+        fig1, ax1 = plt.subplots()
+
+        dms = df[(df['imt'] == imt)]
+        # annual frequency of exceedance:
+        T = from_string(imt).period
+        f = 0 if imt == 0.0 else _find_fact_maxC(T, 'ASCE7-16')
+        imls_o = imtls_dict[imt]
+        imls = [iml*f for iml in imls_o]
+        # have to compute everything for max comp. and for geom. mean
+        RTGM = rtgm_probmce[m]
+        RTGM_o = rtgm_probmce[m]/f
+        afe_target = _find_afe_target(imls, mean_hcurve[m], RTGM)
+        afe_target_o = _find_afe_target(imls_o, mean_hcurve[m], RTGM_o)
+
+        # populate 3-panel plot
+        ax[m].loglog(imls, mean_hcurve[m], 'k', label=_get_label(imt),
+                     linewidth=2, zorder=3)
+        ax[m].loglog([np.min(imls), RTGM], [afe_target, afe_target], 'k--',
+                     linewidth=2, zorder=3)
+        ax[m].loglog([RTGM, RTGM], [0, afe_target], 'k--', linewidth=2,
+                     zorder=3)
+        ax[m].loglog([RTGM], [afe_target], 'ko', label='Probabilistic MCE',
+                     linewidth=2, zorder=3)
+        # populate individual plots
+        ax1.loglog(imls_o, mean_hcurve[m], 'k', label=imt + ' - Geom. mean',
+                   linewidth=2, zorder=3)
+        ax1.loglog([np.min(imls_o), RTGM_o], [afe_target_o, afe_target_o],
+                   'k--', linewidth=2, zorder=3)
+        ax1.loglog([RTGM_o, RTGM_o], [0, afe_target_o], 'k--', linewidth=2,
+                   zorder=3)
+        ax1.loglog([RTGM_o], [afe_target_o], 'ko', label='Probabilistic MCE',
+                   linewidth=2, zorder=3)
+
+        # poes from dms are now rates
+        for ind, (afes, src) in enumerate(zip(dms.poes, dms.src_id)):
+            # get contribution at target level for that source
+            afe_uhgm = _find_afe_target(imls, afes, rtgm_probmce[m])
+            # get % contribution of that source
+            contr_source = afe_uhgm/afe_target
+            out_contr_all.append(contr_source * 100)
+
+        # identify contribution of largest contributor, make color scale
+        largest_contr = np.max(out_contr_all)
+        sample = sum(out_contr_all > fact*largest_contr)
+        viridis = mpl.colormaps['viridis'].reversed()._resample(sample)
+
+        # find and plot the sources, highlighting the ones that contribute more
+        # than 10% of largest contributor
+
+        i = j = 0
```

Comment:
What's the meaning of `i` and `j`? Can we have better names or at least a comment?